WordPress/gutenberg
# Allow negative values for margin inputs

## What?
Allow negative values for margin inputs. Fixes https://github.com/WordPress/gutenberg/issues/32644

## Why?
See https://github.com/WordPress/gutenberg/issues/32644 for details.

## How?
The PR adds a new `spacingType` parameter to the `BoxControl()` function, and allows negative values for the input control in case this parameter is set to "Margin"

## Testing Instructions
- Add to a post a group block
- Try to set the block margin to a negative value
- The input should accept the negative value and the margin should apply to the block
